Core/include/Acts/Propagator/EigenStepper.ipp

```diff
@@ -172,6 +176,31 @@ Acts::Result<double> Acts::EigenStepper<E, A>::step(
     return 0.;
   }
 
+  const auto calcStepSizeScaling = [&](const double errorEstimate_) -> double {
+    // For details about these values see ATL-SOFT-PUB-2009-001 for details
+    constexpr double lower = 0.25;
+    constexpr double upper = 4.0;
+    // This is given by the order of the Runge-Kutta method
+    constexpr double exponent = 0.25;
+
+    // Whether to use fast power function if available
+    constexpr bool tryUseFastPow{false};
+
+    double x = state.options.stepTolerance / errorEstimate_;
+
+    if constexpr (exponent == 0.25 && !tryUseFastPow) {
+      // This is 3x faster than std::pow
+      x = std::sqrt(std::sqrt(x));
+    } else if constexpr (std::numeric_limits<double>::is_iec559 &&
+                         tryUseFastPow) {
+      x = fastPow(x, exponent);
+    } else {
+      x = std::pow(x, exponent);
+    }
+
+    return std::clamp(x, lower, upper);
```

Comment:
Have you considered to clamp before and do the calculation only, when we are between 0.25^4 and 4^4?

Reply:
I didn't try this but I fear that would introduce more code in a very hot path which can make things worse. I don't think we clamp very often. This is more of a failsafe functionality.